### PR TITLE
Update to GraalVM rc12

### PR DESCRIPTION
**Pull request by dsyer**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,6 @@
 target
 *.h
+.classpath
```
Comment (JaroslavTulach) [inline .gitignore:3]: OK.

File: .travis.yml [hunk 2]
```diff
@@ -6,9 +6,9 @@ addons:
 script:
   - env
   - if [ "$_system_name" == "OSX" ]; then S=macos; D=/Contents/Home/; else S=linux; D=/; fi
-  - wget https://github.com/oracle/graal/releases/download/vm-1.0.0-rc7/graalvm-ce-1.0.0-rc7-$S-amd64.tar.gz
-  - tar fxz graalvm-ce-1.0.0-rc7-$S-amd64.tar.gz
-  - JAVA_HOME=graalvm-ce-1.0.0-rc7$D mvn clean install
+  - wget https://github.com/oracle/graal/releases/download/vm-1.0.0-rc12/graalvm-ce-1.0.0-rc12-$S-amd64.tar.gz
```
Comment (JaroslavTulach) [inline .travis.yml:9]: Thanks.

File: pom.xml [hunk 3]
```diff
@@ -13,9 +13,9 @@
             <version>4.5.1</version>
         </dependency>
         <dependency>
-            <groupId>org.graalvm</groupId>
+            <groupId>org.graalvm.sdk</groupId>
             <artifactId>graal-sdk</artifactId>
-            <version>1.0.0-rc5</version>
+            <version>1.0.0-rc12</version>
```
Comment (JaroslavTulach) [inline pom.xml:18]: Good.

File: pom.xml [hunk 4]
```diff
@@ -147,6 +147,60 @@
                 </executions>
             </plugin>
         </plugins>
+        <pluginManagement>
+        	<plugins>
+        		<!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence on the Maven build itself.-->
```
Comment (JaroslavTulach) [inline pom.xml:152]: Fine with me.

File: src/main/java/org/apidesign/demo/matrixultimate/svm/SVMScientificLibraryJNI.java [hunk 5]
```diff
@@ -10,129 +10,118 @@
 import org.graalvm.nativeimage.c.type.CTypeConversion;
 import org.graalvm.word.Pointer;
 
-/** Native image implementation of JNI entry point methods. All the
- * {@link CEntryPoint} methods are on the "boundary" from the native side.
- * The code in here can interface with C libraries without any overhead.
+/**
```
Comment (JaroslavTulach) [inline src/main/java/org/apidesign/demo/matrixultimate/svm/SVMScientificLibraryJNI.java:13]: The changes in this file look like reformatting only. Can we please exclude reformatting related changes from the PR?

**Review by JaroslavTulach — CHANGES_REQUESTED** [review]
Thanks for the contribution. I'll merge it, but please avoid reformatting only changes.

File: src/main/java/org/apidesign/demo/matrixultimate/svm/SVMScientificLibraryJNI.java (hunk 5 above)
Comment (dsyer) [reply src/main/java/org/apidesign/demo/matrixultimate/svm/SVMScientificLibraryJNI.java:13]: I fixed that. Sorry (too many opinions on Java format rules out there - yours are not the same as the other projects I work on apparently).

**Comment by JaroslavTulach** [comment]
The diff looks OK. Alas the builds are failing...

**Comment by dsyer** [comment]
Not my fault I guess (whitespace change)?

**Comment by jtulach** [comment]
I see following error in the log:
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/travis/build/JaroslavTulach/matrixultimate/src/main/java/org/apidesign/demo/matrixultimate/svm/SVMScientificLibraryJNI.java:[26,38] cannot find symbol
  symbol:   variable CreateIsolate
  location: class org.graalvm.nativeimage.c.function.CEntryPoint.Builtin
[INFO] 1 error
```
that doesn't seem like a whitespace change.

**Comment by dsyer** [comment]
I guess it isn't just a whitespace change then. I reverted that file to the original version from the first pull request. It works now, but the whitespace will not be to your liking (my IDE preferences don't match yours). I suggest you either do the upgrade yourself or polish the file that you didn't like when you merge it.

**Comment by JaroslavTulach** [comment]
I'll try to update to RC15 now. Re. formatting - it is not the issue to find the right formatting - the issue is to find an editor that doesn't change existing formatting, I'd say. `vi`, `notepad`, `kate`, `atom`(?), `emacs` could be reasonable alternatives. Editors that use brute force and reformat everything aren't good for cooperation.

**Comment by JaroslavTulach** [comment]
Closing as obsolete as #2 updated to RC15. But thanks anyway.

**Pull request closed without merging**